build/vega-lite-schema.json

```diff
@@ -3888,6 +3888,55 @@
       ],
       "type": "object"
     },
+    "FieldDefWithoutType": {
+      "additionalProperties": false,
+      "properties": {
+        "aggregate": {
+          "$ref": "#/definitions/Aggregate",
+          "description": "Aggregation function for the field\n(e.g., `mean`, `sum`, `median`, `min`, `max`, `count`).\n\n__Default value:__ `undefined` (None)"
+        },
+        "bin": {
```

Comment:
Could `x2` ever be binned differently from `x`?

Reply:
Good point. Actually, we shouldn't even support `bin` for `x2`.  (If `x` is binned, `x2` is inherently the bin end.)

Reply:
Then the name of the field def needs to be something else but I think this will make things much cleaner. So 👍